File: SEP490_G3_MamMoi/frontend/src/API/models/Tree.js
export default class Tree {
  constructor({ treeId, treeName, description, ownerId }) {
    this.treeId = treeId;
    this.treeName = treeName;
    this.description = description;
    this.ownerId = ownerId;
  }

  shortInfo() {
    return `${this.treeName} - Owner #${this.ownerId}`;
  }
}

File: backend/Models/Tree.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MamMoi.Models
{
    [Table("Trees")]
    public class Tree
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TreeId { get; set; }

        [Required]
        [MaxLength(200)]
        public string TreeName { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        // Owner of the tree
        [ForeignKey("Owner")]
        public int OwnerId { get; set; }

        public virtual User? Owner { get; set; }
    }
}
